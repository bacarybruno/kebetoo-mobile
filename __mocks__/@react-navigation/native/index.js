const navigate = jest.fn()
export const useNavigation = () => ({
  navigate,
})

export const useFocusEffect = () => jest.fn()
