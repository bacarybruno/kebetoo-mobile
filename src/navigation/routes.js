const routes = {
  ONBARDING_NAV: 'route:nav:onboarding',
  SIGNIN: 'route:signin',
  SIGNUP: 'route:signup',
  ONBOARDING: 'route:obnoarding',
  HOME_NAV: 'route:nav:home',
  HOME: 'route:home',
  PROFILE: 'route:profile',
  SEARCH: 'route:search',
  STORIES: 'route:stories',
  CREATE_POST: 'route:create_post',
  TABS_FAB: 'route:custom:tabs_fab',
}

export default routes
