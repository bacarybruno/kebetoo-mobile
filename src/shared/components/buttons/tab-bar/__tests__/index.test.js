import ActionButton from 'react-native-action-button'
import { fireEvent } from 'react-native-testing-library'
import { useNavigation } from '@react-navigation/native'

import routes from '@app/navigation/routes'
import setupTest from '@app/config/jest-setup'

import TabBar from '../index'

const givenTabBar = setupTest(TabBar)()

it('renders TabBar', () => {
  const { wrapper } = givenTabBar()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('handles button press', () => {
  const { wrapper } = givenTabBar()
  fireEvent.press(wrapper.root.findByType(ActionButton))
  expect(useNavigation().navigate).toBeCalledTimes(1)
  expect(useNavigation().navigate).toBeCalledWith(routes.CREATE_POST)
})
