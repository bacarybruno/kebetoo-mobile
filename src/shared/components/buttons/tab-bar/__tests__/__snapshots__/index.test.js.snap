// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TabBar 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "zIndex": 1,
        },
        false,
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={null}
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderBottomColor": "rgb(216, 216, 216)",
              "flex": 1,
              "shadowColor": "rgb(216, 216, 216)",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.85,
              "shadowRadius": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "maxHeight": undefined,
            "minHeight": undefined,
            "opacity": undefined,
            "transform": undefined,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityRole="header"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 17,
                  "fontWeight": "600",
                }
              }
            >
              jest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignSelf": "center",
                        "bottom": 12.5,
                        "height": 56,
                        "position": "absolute",
                        "width": 56,
                        "zIndex": 1000,
                      }
                    }
                  >
                    <View
                      pointerEvents="box-none"
                      style={
                        Array [
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "elevation": undefined,
                              "justifyContent": "flex-end",
                              "zIndex": undefined,
                            },
                          ],
                          undefined,
                        ]
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "bottom": 0,
                            "elevation": undefined,
                            "justifyContent": "flex-end",
                            "left": 0,
                            "opacity": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "zIndex": undefined,
                          }
                        }
                      />
                      <View
                        pointerEvents="box-none"
                        style={
                          Array [
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              Object {
                                "elevation": undefined,
                                "justifyContent": "flex-end",
                                "zIndex": undefined,
                              },
                            ],
                            Object {
                              "alignItems": "center",
                            },
                            Object {
                              "paddingVertical": 0,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "marginHorizontal": 0,
                                "zIndex": undefined,
                              },
                              Object {
                                "elevation": 5,
                                "shadowColor": "#000",
                                "shadowOffset": Object {
                                  "height": 5,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.35,
                                "shadowRadius": 3,
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgba(0, 122, 255, 1)",
                                  "borderRadius": 28,
                                  "height": 56,
                                  "width": 56,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "borderRadius": 28,
                                    "height": 56,
                                    "justifyContent": "center",
                                    "transform": Array [
                                      Object {
                                        "scale": 1,
                                      },
                                      Object {
                                        "rotate": "0deg",
                                      },
                                    ],
                                    "width": 56,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "backgroundColor": "transparent",
                                      "color": "rgba(255, 255, 255, 1)",
                                      "fontSize": 30,
                                      "marginTop": -4,
                                    }
                                  }
                                >
                                  +
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
